lib/internal/Magento/Framework/View/Element/Html/Date.php

```diff
@@ -76,20 +93,48 @@ protected function _toHtml()
     public function getEscapedValue()
     {
         if ($this->getFormat() && $this->getValue()) {
-            return strftime($this->getFormat(), strtotime($this->getValue()));
+            return $this->getDateByFormat($this->getFormat(), strtotime($this->getValue()));
         }
         return $this->escapeHtml($this->getValue());
     }
 
     /**
      * Produce and return block's html output
      *
-     * {@inheritdoc}
-     *
      * @return string
      */
     public function getHtml()
     {
         return $this->toHtml();
     }
+
+    /**
+     * Method to get date by format.
+     *
+     * @param string $format
+     * @param int $timestamp
+     *
+     * @return string
+     */
+    private function getDateByFormat(string $format, int $timestamp): string
+    {
+        $format = str_replace(self::DATE_FORMAT_NOT_SUPPORTED, self::DATE_FORMAT_SUPPORTED, $format);
+
+        if (strpos($format, '%') !== false) {
+            $unsupportedData = ['%U', '%V', '%C'];
+
+            foreach ($unsupportedData as $unsupported) {
+                if (strpos($format, $unsupported) !== false) {
+                    if ($unsupported === '%C') {
+                        $format = str_replace($unsupported, round(date("Y", $timestamp) / 100), $format);
+
+                        continue;
+                    }
+                    $format = str_replace($unsupported, date("W", strtotime("-1 day", $timestamp)), $format);
+                }
+            }
+        }
+
+        return date($format, $timestamp);
+    }
```

Comment:
I believe that we should add warning logs that `strftime` formats are deprecated and should be replaced with `date` format

Reply:
Hello @Den4ik 
I am sure, it would be good idea to add this into magento-coding-standard and we have already done it.